actor/cfnetworkingaction/policy.go

```diff
@@ -164,17 +293,15 @@ func (actor Actor) RemoveNetworkPolicy(srcSpaceGUID, srcAppName, destSpaceGUID,
 	return allWarnings, actionerror.PolicyDoesNotExistError{}
 }
 
-func (Actor) transformPolicy(appNameByGuid map[string]string, v1Policy cfnetv1.Policy) Policy {
-	srcName, srcOk := appNameByGuid[v1Policy.Source.ID]
-	dstName, dstOk := appNameByGuid[v1Policy.Destination.ID]
-	if srcOk && dstOk {
-		return Policy{
-			SourceName:      srcName,
-			DestinationName: dstName,
-			Protocol:        string(v1Policy.Destination.Protocol),
-			StartPort:       v1Policy.Destination.Ports.Start,
-			EndPort:         v1Policy.Destination.Ports.End,
-		}
+func (Actor) transformPolicy(appByGuid map[string]v3action.Application, spaceNamesByGUID, orgNamesBySpaceGUID map[string]string, v1Policy cfnetv1.Policy) Policy {
```

Comment:
It took me a while to understand that the new code above (the unique-ing and resolving of GUIDs to structs) is just to pre-build a cache so that you don't need to look up the same org/space/app more than once.

I guess I don't know enough about the likely usage scenarios to know if this optimization is worth the extra code complexity. Do we expect users to have many network policies?

If it is likely that users will have many network policies and will feel the performance pain of looking up the same resource multiple times, maybe this could be structured in a more traditionally caching way. Instead of pre-fetching everything, we could inline `transformPolicy` and build up the caches as we go. Then we also wouldn't need to add the `GetApplicationsByGUIDs` and similar methods to the actor. (I think variadic functions in our actor layer isn't super common in this codebase.)

Reply:
The reason we've added the `GetApplicationsByGUIDs` is because we need to get the applications with only the app guid, we don't know the org or space the application comes from.

The calls for spaces and orgs is necessary because we need to get the space and org names. The API only returns the application's space guid, necessitating multiple calls to get the org name. Effort was made to reduce the number of network calls. We do know that some clients do have a large number of network policies within many spaces/orgs, which would result exponentially more network calls.

Inlining transformPolicy and caching as we go would require a lot more network calls and we do have the information upfront to reduce the number of network calls.

There is potential to de-duplicate and break down large functions. We will take a pass at it.

Reply:
DRY'd up the functions some. Let us know if you see other areas that could be improved.